node/src/blockchain/blockchain_bridge.rs

```
@@ -314,25 +315,31 @@ impl BlockchainBridge {
             .get_block_number()
         {
             Ok(eb) => {
-                if u64::MAX == max_block_count {
+                if u64::MAX == max_block_count || u64::MAX == start_block_nbr {
                     BlockNumber::Number(eb)
                 } else {
                     BlockNumber::Number(eb.as_u64().min(start_block_nbr + max_block_count).into())
                 }
             }
             Err(e) => {
-                info!(
-                    self.logger,
-                    "Using 'latest' block number instead of a literal number. {:?}", e
-                );
```

I think this info log was wrong.

First, I don't like it used to generalise the the two possible situation after that, strictly speaking, we wouldn't be using the 'latest' option, which has usually a different meaning than, "the furthest the blockchain service provider allows us to request at the moment". I'm afraid, if the logs is a good idea at all, we should have two logs for these two different situations.

Second, I understand there has been questions about whether we should care of the error from the blockchain interface, maybe no, but what is definitely wrong is grafting an error message into an info log. The two pieces of information don't belong together.    

I'm pointing to these issues because I'm going to comment on the logging here again and this is important to prevent.

---

My opinion of the logging is that an error from the blockchain service provider doesn't have to be considered an error for the user if the application has a suitable resolution for it. Warning seemed too severe but Info didn't seem like too much. I understand there's a natural mismatch between logging the error at Info level, however, I think it would be worse to just ignore the error in case we receive a response we didn't expect. I can change it to DEBUG and remove the assertions for it.